ts/utils/brightness.ts

```diff
@@ -26,48 +27,67 @@ export const setBrightness = (brightness: number): TaskEither<Error, number> =>
     reason => new Error(String(reason))
   );
 
-const HIGH_BRIGHTNESS = 1.0; // Target screen brightness for a very bright screen
+const HIGH_BRIGHTNESS = 1.0; // Max brightness value
 
 /**
- * Hook that sets the brightness to maximum, until the screen is unmounted
+ * Set the device brightness to the max value and restore the original brightness when
+ * the component is unmount or the app changes state (!== active)
  */
 export const useMaxBrightness = () => {
-  // Brightness effect manager
-  React.useEffect(() => {
-    // eslint-disable-next-line functional/no-let
-    let myBrightness: number | undefined;
+  // The initial device brightness
+  const [startBrightness, setStartBrightness] = useState<number | undefined>(
+    undefined
+  );
+  // The current app state
+  const [appState, setAppState] = useState<AppStateStatus | undefined>(
+    undefined
+  );
+  // Track the current async transition, in order to wait before execute the next async transition
+  const currentTransition = useRef<Promise<void>>(Promise.resolve());
 
-    const myBrightF = async () => {
-      myBrightness = await getBrightness()
-        .fold(
-          () => undefined,
-          _ => _
-        )
-        .run();
-    };
+  AppState.addEventListener("change", setAppState);
 
-    const mySetBrightF = async () => {
-      await myBrightF();
-      if (myBrightness) {
-        await setBrightness(HIGH_BRIGHTNESS).run();
-      }
+  // Change the device brightness
+  const setNewBrightness = async (brightness: number) => {
+    await currentTransition.current;
+    await setBrightness(brightness).run();
+  };
+
+  // First mount, read and save the current device brightness
+  React.useEffect(() => {
+    const getCurrentBrightness = async () => {
+      setStartBrightness(
+        await getBrightness()
+          .fold(
+            () => undefined,
+            _ => _
+          )
+          .run()
```

Comment:
I admit I'm having hard times to understand this code 🙇  
It seems overly complex for what it is doing:

1. `getBrightness` wraps `getBrightnessPlatform` in a TaskEither
2. the only use of `getBrightness` (besides [a spy in FiscalCodeLandscapeOverlay](https://github.com/pagopa/io-app/blob/master/ts/components/__tests__/FiscalCodeLandscapeOverlay.test.tsx#L55)) is this one, where we ignore the error case

Wouldn't the code below be equivalent while being simpler to understand (two chained promises)?

```typescript
// silently discard the error, set the initial brightness if available, Promise still flows down
const getCurrentBrightness = getBrightness().catch(_ => undefined).then(setStartBrightness)
```

Reply:
You're right, I focused on fixing the bug without refactoring the existing code. 
In 173e352a83abed015c02a81e90a0327a55c8665c I've removed the unnecessary wrapping and refactoring the related code.
Btw despite now `getBrightness` and `setBrightness` are not exported, is always possible, from the outside, use directly the library `react-native-screen-brightness` and the encapsulation is not guaranteed.